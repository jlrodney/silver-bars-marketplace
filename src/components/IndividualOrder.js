import React from 'react';

const IndividualOrder = (props) => (
  <div>
    <p>{props.type}: {props.quantity} kg for Â£{props.price} [{props.userId}]
      <button className="removeOrderButton"
        onClick={(e) => {
          props.handleDeleteOrder([props.orderId, props.type]);
        }}
      >
      remove
      </button>
    </p>
  </div>
);

export default IndividualOrder;
