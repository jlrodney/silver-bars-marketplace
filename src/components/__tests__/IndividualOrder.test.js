import React from 'react';
import { shallow } from 'enzyme';
import IndividualOrder from '../IndividualOrder';

const order = {
    orderId: 'xyz986',
    userId: '1',
    quantity: 22,
    price: 230,
    type: 'BUY'
  }
describe('Order item', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<IndividualOrder
      key={order.orderId}
      orderId={order.orderId}
      userId={order.userId}
      quantity={order.quantity}
      price={order.price}
      type={order.type}
    />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should have text associated with order', () => {
    const wrapper = shallow(<IndividualOrder
      key={order.orderId}
      orderId={order.orderId}
      userId={order.userId}
      quantity={order.quantity}
      price={order.price}
      type={order.type}
    />);
    expect(wrapper.find('p').text()).toEqual("BUY: 22 kg for Â£230 [1]remove");
  });
  it('should call onClick prop for invalid form submission', () => {
    const onHandleDeleteOrderSpy = jest.fn();
    const wrapper = shallow(<IndividualOrder
      key={order.orderId}
      orderId={order.orderId}
      userId={order.userId}
      quantity={order.quantity}
      price={order.price}
      type={order.type}
      handleDeleteOrder={onHandleDeleteOrderSpy}
    />);
    wrapper.find('button').simulate('click');
    expect(onHandleDeleteOrderSpy).toBeCalled();
  });
});
