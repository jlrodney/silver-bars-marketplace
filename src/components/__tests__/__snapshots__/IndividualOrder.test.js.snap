// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expense item should render correctly 1`] = `
<div>
  <p>
    BUY
    : 
    22
     kg for Â£
    230
     [
    1
    ]
    <button
      className="removeOrderButton"
      onClick={[Function]}
    >
      remove
    </button>
  </p>
</div>
`;
